packages/client/src/getPools.ts

```diff
@@ -1,53 +1,88 @@
 import UniswapV2Factory from "@uniswap/v2-core/build/UniswapV2Factory.json";
 import UniswapV2Pair from "@uniswap/v2-core/build/UniswapV2Pair.json";
 import { getAddress, ParamChainName } from "@zetachain/protocol-contracts";
+import SystemContract from "@zetachain/protocol-contracts/abi/zevm/SystemContract.sol/SystemContract.json";
 import { ethers } from "ethers";
 
 import { ZetaChainClient } from "./client";
 
 export const getPools = async function (this: ZetaChainClient) {
   const rpc = this.getEndpoint("evm", `zeta_${this.network}`);
   const provider = new ethers.providers.StaticJsonRpcProvider(rpc);
-
-  const uniswapV2FactoryAddress = getAddress(
-    "uniswapV2Factory",
-    `zeta_${this.network}` as ParamChainName
-  );
+  const zetaNetwork = `zeta_${this.network}` as ParamChainName;
+  const uniswapV2FactoryAddress = getAddress("uniswapV2Factory", zetaNetwork);
 
   if (!uniswapV2FactoryAddress) {
     throw new Error("uniswapV2Factory is not defined");
   }
 
-  const UniswapV2FactoryContract = new ethers.Contract(
-    uniswapV2FactoryAddress,
-    UniswapV2Factory.abi,
+  const systemContractAddress = getAddress("systemContract", zetaNetwork);
+  if (!systemContractAddress) {
+    throw new Error("System contract is not defined");
+  }
+
+  const systemContract = new ethers.Contract(
+    systemContractAddress,
+    SystemContract.abi,
     provider
   );
 
-  const totalPairs = await UniswapV2FactoryContract.allPairsLength();
-  let pairs = [];
-  for (let i = 0; i < totalPairs; i++) {
-    pairs.push(await UniswapV2FactoryContract.allPairs(i));
+  const zetaTokenAddress = getAddress("zetaToken", zetaNetwork);
+  if (!zetaTokenAddress) {
+    throw new Error("ZETA token address is not defined");
   }
 
-  const poolPromises = pairs.map(async (pair: any) => {
-    let pool = {
-      pair,
-      t0: {},
-      t1: {},
-    } as any;
-    const pairContract = new ethers.Contract(pair, UniswapV2Pair.abi, provider);
+  const foreignCoins = await this.getForeignCoins();
+  const tokenAddresses = foreignCoins.map(
+    (coin: any) => coin.zrc20_contract_address
+  );
+  tokenAddresses.push(zetaTokenAddress);
+
+  const poolPromises = [];
 
-    pool.t0.address = await pairContract.token0();
-    pool.t1.address = await pairContract.token1();
+  for (let i = 0; i < tokenAddresses.length; i++) {
+    for (let j = i + 1; j < tokenAddresses.length; j++) {
```

Comment:
This will generate the permutations for `{tokenA, tokenB}` and `{tokenB, tokenA}`, not sure if we want to skip one of those if they're the same pool

Reply:
Removed non-unique combinations and made it more functional: 65e3013